### PR TITLE
[Feature] Update dependencies

### PR DESCRIPTION
**Pull request by SferaDev**
### :memo: Implementation

- Do not crash app if data element is not included in program (Country support issue): [Reference](https://github.com/EyeSeeTea/Bulk-Load/pull/205/files#diff-f0776af6b1cda00ae465590e1e4064b7829b3772fcbb103a0e703c760d42e464R662-R665)
- Update dependencies and configuration
- Prettify code

### CODE REVIEW

**Comment by adrianq** [comment]
@tokland As part of the next release in SP we will be updating the dependencies and libraries. Can you please review this PR and use project monitoring to test it? We will be releasing a new version of Bulk Load by Wednesday so if you can have a look at it on Monday or Tuesday it would be great. You can already create the PR for project monitoring if you wish. There is already a card for this in clickup

File: .eslintrc.js [hunk 1]
```diff
@@ -37,6 +37,8 @@ module.exports = {
         "@typescript-eslint/type-annotation-spacing": "off",
         "@typescript-eslint/no-use-before-define": "off",
         "@typescript-eslint/explicit-function-return-type": "off",
+        "no-use-before-define": "off",
```
Comment (tokland) [inline .eslintrc.js:40]: Not in this diff, but it seems `@typescript-eslint/no-use-before-define` is written twice.

File: .eslintrc.js [hunk 2]
```diff
@@ -3,19 +3,19 @@
...
         "react-app",
-        "prettier",
         "eslint:recommended",
         "plugin:react/recommended",
         "plugin:cypress/recommended",
         "plugin:@typescript-eslint/recommended",
-        "prettier/@typescript-eslint",
-        "plugin:prettier/recommended",
     ],
     rules: {
```
Comment (tokland) [inline .eslintrc.js:11]: When I run "yarn link" I get 5 errors about rule "no-redeclare" . Are you seeing those?

**Review by tokland — COMMENTED** [review]
All good, just a minor comment on .eslintrc. [update] I found that yarn lint throws some errors, can you confirm?

File: .eslintrc.js (hunk 2 above)
Comment (SferaDev) [reply .eslintrc.js:11]: Fixed, the cypress configuration files had redeclaration of global variables as the cy object... I've already had to remove those from other projects in the past (training-app for example).

File: .eslintrc.js (hunk 1 above)
Comment (SferaDev) [reply .eslintrc.js:39]: Removed the duplicate entry

**Comment by SferaDev** [comment]
Updated PR with couple of commits I had on the work in progress branch:

- Update composition root to match the rest of our projects (this was the first one and was still a Singleton class)
    - This fixes development fast refresh from latest CRA too
- Update cookie flag for cypress (let's see if travis begins passing again)
- Update tests to transform ``@eyeseetea/d2-ui-components`` ES5 imports
- Update tests to include ``Snackbar`` and ``Loading`` contexts

**Review by tokland — COMMENTED** [review]
@SferaDev , Can you reproduce this?

Settings -> Change any setting:
``` 
Error: Objects are not valid as a React child (found: TypeError: Cannot read property 'settings' of undefined)
```

**Comment by adrianq** [comment]
I can also reproduce this error:
image

**Review by tokland — APPROVED** [review]
A couple of commits regarding a problem with a compositionRoot missing, and now it seems to be working fine in some tests I made. Let's check some time why ts/eslint allowed that "any" in there. 

So fine by me :+1:

**Pull request merged**